osquery/tables/networking/aws_metadata.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright (c) Jean-Francois Dive <jfdive@gmail.com>
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. 
+ *
+ */
+
+#include <string>
+
+#include <boost/network/protocol/http/client.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/tables/networking/aws_metadata.h"
+
+namespace pt = boost::property_tree;
+namespace http = boost::network::http;
+
+namespace osquery {
+namespace tables {
+
+//#define AWS_META_DEBUG true
+
+static bool initialized = false;
+
+// use of the ipv4 address to be region independent (vs hostname based)
+const static std::string awsBaseUrl{"http://169.254.169.254/latest/"};
+
+static std::vector<AwsData*> AwsFields; 
```

Comment:
Problem is .. and maybe i am wrong on this, that you cant have a vector<AwsData> filled with its defined sub-classes. I made a few tests programs to play around this and as long as i have a pure virtual the compiler will refuse to define std::vector<the_base>. What am i missing ? I am attaching a test cpp file i used 
[testcpp.txt](https://github.com/facebook/osquery/files/693642/testcpp.txt)
[testcpp.txt](https://github.com/facebook/osquery/files/693643/testcpp.txt)